File: hw4/HW4S20/mm.py
import sys

#open files from command line args
data = open(sys.argv[1])
template = open(sys.argv[2])

#put lines of data file into a list
lines = data.readlines()

#split the first line at the commas and subtract the last newline char
headers = lines[0][0:len(lines[0])-1].split(',')

#extract template from file
temp = template.read()

#make dictionary of all columns
cols = {}
for i in range(0, len(headers)):
    cols[headers[i]] = i

#loop through each line of CSV file
for i in range(1, len(lines)):
    line = lines[i][0:len(lines[i])-1].split(',')
    filename = line[cols["ID"]] + ".txt"
    userFile = open(filename, "w")
    result = temp
    for col in cols.keys():
        index = result.find("<<" + col + ">>")
        length = len("<<" + col + ">>")
        result = result[0:index] + line[cols[col]] + result[index+length:]
    userFile.write(result)

File: hw4/HW4S20/MM.cs
using System;
using System.IO;
using System.Linq;
namespace HW4 {
    public class MM {
        public static void Main(string[] args) {
            StreamReader input = new StreamReader(args[0]);
            string[] headers = input.ReadLine().Split(',');
            Dictionary<string, int> cols = new Dictionary<string, int>();
            for (int i = 0; i < headers.Length; i++) {
                cols.Add(headers[i], i);
            }
            List<string[]> data = new List<string[]>();
            while(!input.EndOfStream) {
                string line = input.ReadLine();
                data.Add(line.Split(','));
            }
            input.Close();
            input = new StreamReader(args[1]);
            string temp = input.ReadToEnd();
            for (int i = 0; i < data.Count; i++) {
                StreamWriter output = new StreamWriter(data[i][cols["ID"]] + ".txt");
                string result = temp;
                foreach (var column in cols) {
                    string search = "<<"+column.Key+">>";
                    int index = result.IndexOf(search);
                    int length = search.Length;
                    result = result.Substring(0,index) + data[column.Value] + result.Substring(index+length);
                }
                output.Write(result);
            }
        }
    }
}